client/js/is-sum-formula.js

```diff
@@ -0,0 +1,40 @@
+const isSumFormula = function(string, numCols, numRows) {
+	// return [col, startRow, endRow] if string is a sum function
+	// otherwise return false
+
+	// string must be in form
+	// =SUM([One capital letter][1 or more digits]:[One capital letter][1 or more digits])
+	const re = /=SUM\(([A-Z]\d+):([A-Z]\d+)\)$/;
+
+	// if string is in valid format, get col letters and row numbers
+	if (string.match(re) !== null) {
+
+		const numberPattern = /\d+/g;
+		const letterPattern = /[A-Z]/g;
+
+		const colNumbers = string.slice(4) // don't get letters in "SUM"
+		                    .match(letterPattern)
+		                    .map(x => x.charCodeAt(0) - 64); 
+		const rowNumbers = string.match(numberPattern)
```

Comment:
it's kinda weird to separately match column letters and row numbers. 

You saw above that you can capture the column (e.g., `A`) or cell (e.g., `A15`) definition as a match group.

You can make a tiny helper function that accepts such a value and returns a (col, row) value. Since you designed the system to only have A-Z columns, you can just pivot on the first character like this: 

const location = 'A15';
const col = location[0]; // will be `A` in this example
const row = location.slice(1); // will `15` in this example. Would be blank string if this location is only the column, e.g., `A`